src/libcore/macros.rs

```diff
@@ -15,7 +15,8 @@ macro_rules! panic {
         panic!("explicit panic")
     );
     ($msg:expr) => ({
-        static _MSG_FILE_LINE: (&'static str, &'static str, usize) = ($msg, file!(), line!());
+        static _MSG_FILE_LINE: (&'static str, &'static str, usize) =
```

Comment:
Could you use `#[cfg(stage0)]` and `#[cfg(not(stage0))]` to provide usize and u32 versions and thus avoid the errors?